model/eventencoder/v2.go

```diff
@@ -0,0 +1,56 @@
+package eventencoder
+
+import (
+	"github.com/gotify/server/model"
+)
+
+// V2 encodes given events within the constraints of API version 1.
+type V2 struct {
+	Version uint
+}
+
+// NewV2 creates a new instance of V2.
+func NewV2() *V2 {
+	return &V2{
+		2,
+	}
+}
+
+// Encode encodes the given event.
+func (encoder V2) Encode(event model.Event) (interface{}, error) {
+	var encoded eventMessageV2
+
+	switch event := event.(type) {
+	case *model.Message:
+		encoded.Type = "message"
+		encoded.Content = event.ToExternal()
+	case *model.MessageDeletions:
+		encoded.Type = "message_deletions"
+		encoded.Content = event.ToExternal()
+	default:
+		err := newTypeNotSupportedError(event, encoder.Version)
+		return nil, err
+	}
+
+	return encoded, nil
+}
+
+// eventMessageV2 Model
+//
+// eventMessageV2 holds information about a message which will be sent to the clients.
+//
+// swagger:model eventMessageV2
+type eventMessageV2 struct {
+	// The event type.
+	//
+	// read only: true
+	// required: true
+	// example: message_deletion
+	Type string `json:"type"`
+	// The event content.
+	//
+	// read only: true
+	// required: true
+	// example: [14,15,16]
+	Content interface{} `json:"content"`
```

Comment:
Maybe the content of the events should be included in the parent object. Then we could use the discriminator feature from openapi v3 (https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/ see discriminator).

So the message would look like this:
```
{ 
    type: "message",
    message: "my content",
    title: "...",
    ...
}
```

This would allow us to f.ex. generate the java client with proper jackson bindings for serializing the message into the right event type.

Reply:
This sounds promising indeed. I'd like to first have the Swagger specification pass the tests, so I can adjust the specification from there. Any help getting the Travis CI test pass is very appreciated.

Reply:
Sadly, we currently use openapi v2 /swagger 2 so the change to openapi v3 would probably done in a separate pr.

Reply:
Isn't [this](https://goswagger.io/use/spec/discriminated.html) exactly what we need?

Reply:
Yeah it is :D. I thought it was only possible in openapi v3.